cloud/services/loadbalancers.go

```diff
@@ -110,11 +110,6 @@ func AddNodeToNB(
 	logger logr.Logger,
 	machineScope *scope.MachineScope,
 ) error {
-	// Update the NB backend with the new instance if it's a control plane node
-	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
-		return nil
-	}
```

Comment:
nb: I moved this check to the caller. We don't expect the function to be used elsewhere correct?